sim/midas/src/main/cc/simif.cc

```diff
@@ -14,6 +14,83 @@ double diff_secs(midas_time_t end, midas_time_t start) {
   return ((double)(end - start)) / TIME_DIV_CONST;
 }
 
+void simulation_t::record_start_times() {
+  start_hcycle = sim.actual_hcycle();
+  start_time = timestamp();
+}
+
+void simulation_t::record_end_times() {
+  end_time = timestamp();
+  end_tcycle = sim.actual_tcycle();
+  end_hcycle = sim.actual_hcycle();
+}
+
+void simulation_t::print_simulation_performance_summary() {
+  // Must call record_start_times and record_end_times before invoking this
+  // function
+  assert(start_hcycle != -1);
+  assert(end_hcycle != 0);
+  uint64_t hcycles = end_hcycle - start_hcycle;
+  double sim_time = diff_secs(end_time, start_time);
+  double sim_speed = ((double)end_tcycle) / (sim_time * 1000.0);
+  double measured_host_frequency = ((double)hcycles) / (sim_time * 1000.0);
+  double fmr = ((double)hcycles / end_tcycle);
```

Comment:
```suggestion
  const uint64_t hcycles = end_hcycle - start_hcycle;
  const double sim_time = diff_secs(end_time, start_time);
  const double sim_speed = ((double)end_tcycle) / (sim_time * 1000.0);
  const double measured_host_frequency = ((double)hcycles) / (sim_time * 1000.0);
  const double fmr = ((double)hcycles / end_tcycle);
```

I think const correctness dictates that these should be const